runner/src/main.rs

```diff
@@ -146,7 +135,11 @@ fn run_examples(opt: &RunExamples) -> Step {
             /// top-level method first.
             steps: examples
                 .iter()
-                .map(|example| run_example(example))
+                .filter(|example| match &opt.example_name {
+                    Some(example_name) => &example.name == example_name,
+                    None => true,
+                })
+                .map(|example| run_example(opt, example))
```

Comment:
As explained in my comment on the `run_example` func, I think it might be nicer to pass just the `build_only` key, instead of the whole opt struct. :)

Reply:
We will also need access to the application variant from within `run_example` at some point, plus I think most other arguments to that struct (e.g. whether to compile in debug mode) so I think it is fine to just pass the entire thing. It acts kind of as a context of how to run the examples.

Reply:
That makes sense. Perhaps it's more akin to a closure, than the API of a reusable function then. :)